### :x: 1 Tests Failed:
| Tests completed | Failed | Passed | Skipped |
|---|---|---|---|
| 2800 | 1 | 2799 | 29 |
<details><summary>View the full list of 1 :snowflake: flaky tests</summary>

> 
> ```
> github.com/status-im/status-go/protocol TestMessengerCommunitiesTokenPermissionsSuite
> ```
> 
> **Flake rate in main:** 96.22% (Passed 16 times, Failed 407 times)
> <details><summary>Stack Traces | 87.9s run time</summary>
> 
> > 
> > ```
> > === RUN   TestMessengerCommunitiesTokenPermissionsSuite&#xA;2024-09-27T15:46:31.182Z&#x9;DEBUG&#x9;p2p-config&#x9;config/log.go:21&#x9;[Fx] HOOK OnStop&#x9;&#x9;github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2.1() executing (caller: github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2)&#xA;2024-09-27T15:46:34.429Z&#x9;DEBUG&#x9;p2p-config&#x9;config/log.go:21&#x9;[Fx] HOOK OnStop&#x9;&#x9;github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2.1() called by github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2 ran successfully in 4.168µs&#xA;2024-09-27T15:46:37.300Z&#x9;DEBUG&#x9;p2p-config&#x9;config/log.go:21&#x9;[Fx] HOOK OnStop&#x9;&#x9;github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2.1() called by github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2 ran successfully in 6.001µs&#xA;2024-09-27T15:47:37.957Z&#x9;DEBUG&#x9;p2p-config&#x9;config/log.go:21&#x9;[Fx] HOOK OnStop&#x9;&#x9;github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2.1() called by github..../libp2p/go-libp2p/config.(*Config).addAutoNAT.func2 ran successfully in 14.317µs&#xA;--- FAIL: TestMessengerCommunitiesTokenPermissionsSuite (87.89s)&#xA;
> > ```
> 
> </details>

</details>

To view individual test run time comparison to the main branch, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/status-im/status-go/tests/fix%2Fno-revealed-accounts-in-response)

--- a/protocol/communities/manager.go
+++ b/protocol/communities/manager.go
@@ -4897,8 +4897,8 @@ func (m *Manager) ShareRequestsToJoinWithPrivilegedMembers(community *Community,
 	return nil
 }
 
-func (m *Manager) shareAcceptedRequestToJoinWithPrivilegedMembers(community *Community, requestsToJoin *RequestToJoin) error {
-	pk, err := common.HexToPubkey(requestsToJoin.PublicKey)
+func (m *Manager) shareAcceptedRequestToJoinWithPrivilegedMembers(community *Community, requestToJoin *RequestToJoin) error {
+	pk, err := common.HexToPubkey(requestToJoin.PublicKey)
 	if err != nil {
 		return err
 	}
@@ -4906,9 +4906,13 @@ func (m *Manager) shareAcceptedRequestToJoinWithPrivilegedMembers(community *Com
 	acceptedRequestsToJoinWithoutRevealedAccounts := make(map[string]*protobuf.CommunityRequestToJoin)
 	acceptedRequestsToJoinWithRevealedAccounts := make(map[string]*protobuf.CommunityRequestToJoin)
 
-	acceptedRequestsToJoinWithRevealedAccounts[requestsToJoin.PublicKey] = requestsToJoin.ToCommunityRequestToJoinProtobuf()
-	requestsToJoin.RevealedAccounts = make([]*protobuf.RevealedAccount, 0)
-	acceptedRequestsToJoinWithoutRevealedAccounts[requestsToJoin.PublicKey] = requestsToJoin.ToCommunityRequestToJoinProtobuf()
+	acceptedRequestsToJoinWithRevealedAccounts[requestToJoin.PublicKey] = requestToJoin.ToCommunityRequestToJoinProtobuf()
+
+	revealedAccounts := requestToJoin.RevealedAccounts
+	requestToJoin.RevealedAccounts = make([]*protobuf.RevealedAccount, 0)
+	acceptedRequestsToJoinWithoutRevealedAccounts[requestToJoin.PublicKey] = requestToJoin.ToCommunityRequestToJoinProtobuf()
+	// Set back the revealed accounts
+	requestToJoin.RevealedAccounts = revealedAccounts
 
 	msgWithRevealedAccounts := &protobuf.CommunityPrivilegedUserSyncMessage{
 		Type:          protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN,

--- a/protocol/communities_messenger_test.go
+++ b/protocol/communities_messenger_test.go
@@ -922,6 +922,8 @@ func (s *MessengerCommunitiesSuite) TestRequestAccess() {
 	response, err = s.bob.AcceptRequestToJoinCommunity(acceptRequestToJoin)
 	s.Require().NoError(err)
 	s.Require().NotNil(response)
+	s.Require().NotEmpty(response.RequestsToJoinCommunity())
+	s.Require().Len(response.RequestsToJoinCommunity()[0].RevealedAccounts, 1)
 
 	s.Require().Len(response.Communities(), 1)
 

